lib/hal/video.c

```diff
@@ -310,67 +335,26 @@ void XVideoInit(DWORD dwMode, int width, int height, int bpp)
 
 BOOL XVideoSetMode(int width, int height, int bpp, int refresh)
 {
-	VIDEO_MODE_SETTING *pVidMode = NULL;
-	int vidRefresh = 0;
-	int i = 0;
-
-	DWORD dwEnc = XVideoGetEncoderSettings();
-	
-	DWORD dwAdapter		= dwEnc & 0x000000FF;
-	DWORD dwStandard	= dwEnc & 0x0000FF00;
-
-	if(bpp != 15 && bpp != 16 && bpp != 32)
-		bpp = 32;
-
-	if(refresh > 0)
-		vidRefresh = refresh;
-	else
-	{
-		if(dwEnc & 0x00400000) //60Hz refresh rate
-			vidRefresh = 60;
-		else
-			vidRefresh = 50;
-	}
+	VIDEO_MODE vm;
+	void *p = NULL;
 
-	for(i=0; i<iVidModes; i++)
+	while(XVideoListModes(&vm, bpp, refresh, &p))
 	{
-		pVidMode = &vidModes[i];
-
-		if((pVidMode->dwFlags & 0x000000FF) != dwAdapter)
-			continue;
-
-		if(pVidMode->dwStandard != dwStandard)
-			continue;
-
-		if(pVidMode->width != width || pVidMode->height != height)
-			continue;
-
-		if(pVidMode->refresh != vidRefresh)
-			continue;
+		if(vm.width == width && vm.height == height)
+		{
```

Comment:
One could re-write this as:

```c
if(pVidMode->width != width || pVidMode->height != height)
	continue;
```

This makes it closer to code in `XVideoListModes` which is good when we move code around.
Also avoids indentation issues when more conditions are added.

Reply:
I can't quite figure out how to make this change work at this hour. Will have to revisit - will post a comment in the upstream PR unless you beat me to it.